plugins/catalog-react/src/components/EntityKindPicker/EntityKindPicker.tsx

```diff
@@ -28,8 +28,11 @@ export const EntityKindPicker = ({
   initialFilter,
   hidden,
 }: EntityKindFilterProps) => {
-  const [selectedKind] = useState(initialFilter);
-  const { updateFilters } = useEntityListProvider();
+  const { updateFilters, queryParameters } = useEntityListProvider();
+  const [selectedKind] = useState(
```

Comment:
Since query param values can also be an array, maybe a helper function(s) to translate multiple values to a single value would help (and vice versa). Not sure the plan moving forward for multi values, but I could see this needed at some point anyways?

Reply:
I would recommend to use the `qs` package with the default options. That handles all cases. We already have an implementation here: https://github.com/backstage/backstage/blob/master/packages/core/src/hooks/useQueryParamState.ts